presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -933,32 +1387,60 @@ public void testFilterContainmentWithMismatchStringLength()
     {
         String originalViewSql = format("SELECT a, b FROM %s WHERE b <> 'banana'", BASE_TABLE_6);
         String baseQuerySql = format("SELECT a, b FROM %s WHERE b = 'apple'", BASE_TABLE_6);
-        String expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 'apple'", VIEW);
+        String expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 'apple'", VIEW_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
 
         originalViewSql = format("SELECT a, b FROM %s WHERE b NOT IN ('USA','CAN')", BASE_TABLE_6);
         baseQuerySql = format("SELECT a, b FROM %s WHERE b = 'UK'", BASE_TABLE_6);
-        expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 'UK'", VIEW);
+        expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 'UK'", VIEW_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
     }
 
-    private void assertOptimizedQuery(String originalViewSql, String baseQuerySql, String expectedViewSql)
+    private void assertOptimizedQuery(String baseQuerySql, String expectedQueryAfterOptimization, Map<String, List<String>> viewQueries)
     {
-        Table viewTable = new Table(QualifiedName.of(VIEW));
-
-        Query originalViewQuery = (Query) SQL_PARSER.createStatement(originalViewSql, PARSING_OPTIONS);
-        Query baseQuery = (Query) SQL_PARSER.createStatement(baseQuerySql, PARSING_OPTIONS);
-        Query expectedViewQuery = (Query) SQL_PARSER.createStatement(expectedViewSql, PARSING_OPTIONS);
-
         transaction(transactionManager, accessControl)
                 .singleStatement()
                 .readUncommitted()
                 .readOnly()
-                .execute(CLIENT_SESSION, session -> {
-                    Query optimizedBaseToViewQuery = (Query) new MaterializedViewQueryOptimizer(metadata, session, SQL_PARSER, accessControl, domainTranslator, viewTable, originalViewQuery).rewrite(baseQuery);
+                .execute(TEST_SESSION, session -> {
+                    Query baseQuery = (Query) SQL_PARSER.createStatement(baseQuerySql, createParsingOptions(session));
+                    Query expectedViewQuery = (Query) SQL_PARSER.createStatement(expectedQueryAfterOptimization, createParsingOptions(session));
+                    Map<QualifiedObjectName, List<ConnectorMaterializedViewDefinition>> baseTableToMaterializedViewMap = new HashMap<>();
+
+                    int i = 1;
```

Comment:
We can avoid viewcount if we send like: 

ImmutableMap.of(
                BASE_TABLE_1, ImmutableMap.of(VIEW_1, viewSql1))